projects/spectator/src/lib/internals/element-focus.ts

```diff
@@ -1,13 +1,69 @@
 import { dispatchFakeEvent } from '../dispatch-events';
+import { isRunningInJsDom } from '../utils';
+
+/** Property name added to HTML Elements to ensure we don't double-patch focus methods on an element. */
+const IS_FOCUS_PATCHED_PROP = '_patched_focus';
+
+/** Ensures that a single set of matching focus and blur events occur when HTMLElement.focus() is called. */
+class FocusEventWatcher implements EventListenerObject {
+
+  private readonly _active: Element | null;
+  private _blurred = false;
+  private _focused = false;
+
+  constructor(private readonly _e: HTMLElement) {
+    this._active = _e.ownerDocument.activeElement;
+    this._e.addEventListener('focus', this);
+    this._active?.addEventListener('blur', this);
+  }
+  public handleEvent(evt: Event): void {
+    if (evt.type === 'focus') {
```

Comment:
```ts
this._focused = type === 'focus';
this._blurred = type === 'blur';
```

Reply:
That's incorrect - we are listening for both a blur event and a focus event on different elements, that would overwrite the first event when the second event occurs.